app/src/main/java/io/apicurio/registry/metrics/RestMetricsInterceptor.java

```diff
@@ -0,0 +1,85 @@
+package io.apicurio.registry.metrics;
+
+import org.eclipse.microprofile.metrics.ConcurrentGauge;
+import org.eclipse.microprofile.metrics.Counter;
+import org.eclipse.microprofile.metrics.Metadata;
+import org.eclipse.microprofile.metrics.MetricRegistry;
+import org.eclipse.microprofile.metrics.Tag;
+import org.eclipse.microprofile.metrics.Timer;
+import org.eclipse.microprofile.metrics.annotation.RegistryType;
+
+import javax.inject.Inject;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+
+import static io.apicurio.registry.metrics.MetricIDs.*;
+import static org.eclipse.microprofile.metrics.MetricRegistry.Type.APPLICATION;
+import static org.eclipse.microprofile.metrics.MetricType.*;
+import static org.eclipse.microprofile.metrics.MetricUnits.MILLISECONDS;
+
+/**
+ * Interceptor that tracks metrics across all REST resources.
+ *
+ * @author Jakub Senko <jsenko@redhat.com>
+ */
+@Interceptor
+@RestMetricsApply
```

Comment:
Surprised to see this here.

Reply:
Where else is this annotation used and also where is it processed?  I can't figure out who processes this annotation.  Not sure how I missed that part of the review...

Reply:
This annotation binds this interceptor to be used on REST resource classes e.g https://github.com/Apicurio/apicurio-registry/pull/241/files?file-filters%5B%5D=.java#diff-f9081a654b4a4101633c7b7b1b7caf1fR78

It's collecting metrics across all of them, since the standard metrics annotations can only do this per-class/method.
It's possible that such aggregated metrics can be created on the receiver's side e.g. Prometheus, but this could be a way to provide smaller number of more useful metrics.

Reply:
So I guess it's the platform that performs that binding?  The annotation doesn't have to be registered in any way?  Also, the definition of the annotation `@RestMetricsApply` indicates it can be used on a class or on a method but it's only used (I think) on classes.

Reply:
Note: I misunderstood, the binding is done via `@javax.interceptor.InterceptorBinding`